src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
On 09/06/2016 00:02, Mathieu Duponchelle wrote:
> I figured out what the actual issue was here, and am not sure how to best
> solve it:
>
> The "core extensions" are not statically linked with the executable, but
> dlopened, however we are passing -fvisibility=hidden to the compiler, and
> DEFAULT_MEM_ALLOCATOR ends up having a different address in cmark_node_new
> when called from the dlopened library.

You should probably call `dlopen` with the `RTLD_GLOBAL` flag.